manifests/settings_file.pp

```diff
@@ -12,13 +13,31 @@
 # mode: settings file's mode
 #
 define foreman_proxy::settings_file (
-    $path = "/etc/foreman-proxy/settings.d/${title}.yml",
+    $enabled       = undef,
+    $listen_on     = undef,
+    $path          = "/etc/foreman-proxy/settings.d/${title}.yml",
+    $owner         = 'root',
+    $group         = $::foreman_proxy::user,
+    $mode          = '0640',
     $template_path = "foreman_proxy/${title}.yml.erb",
-    $owner = 'root',
-    $group = $::foreman_proxy::user,
-    $mode = '0640',
   ) {
 
+  if $title != 'settings' {
+    $real_enabled = pick($enabled, getvar("foreman_proxy::${title}"))
+    $real_listen_on = pick($listen_on, getvar("foreman_proxy::${title}_listen_on"))
+
```

Comment:
I kind of like the `foreman_settings` define.  How about making the if cleaner by having a boolean for whether we're configuring a module or not? See latest.